aplt/vapid.py

```diff
@@ -0,0 +1,120 @@
+import base64
+import time
+import hashlib
+
+import ecdsa
+import logging
+from jose import jws
+
+
+class VapidException(Exception):
+    pass
+
+
+class Vapid(object):
+    """Minimal VAPID signature generation library. """
+    _private_key = None
+    _public_key = None
+    _hasher = hashlib.sha256
+
+    def __init__(self, private_key_file=None, private_key=None):
+        """Initialize VAPID using an optional file containing a private key
+        in PEM format.
+
+        :param private_key_file: The name of the file containing the
+        private key
+        """
+        if private_key_file:
+            private_key = open(private_key_file).read()
```

Comment:
The file handle isn't closed by default from the scope escaping. A with statement is usually used for reading as it ensures the file handle is closed.